src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java

```diff
@@ -207,25 +212,19 @@ private static String getEndpointSuffix(String blobURL)
      * @throws StorageException
      * @throws IOException
      * @throws InterruptedException
+	 * @returns Md5 hash of the uploaded file in hexadecimal encoding
      */
-    protected static void upload(TaskListener listener, CloudBlockBlob blob, FilePath src)
-	    throws StorageException, IOException, InterruptedException {
-	long startTime = System.currentTimeMillis();
-	InputStream inputStream = src.read();
-        
-	try {
-	    blob.upload(inputStream, src.length(), null,
-		    getBlobRequestOptions(), Utils.updateUserAgent());
-            
-	} finally {
-	    try {
-		inputStream.close();
-	    } catch (IOException e) {
-
-	    }
-	}
-	long endTime = System.currentTimeMillis();
-	listener.getLogger().println("Uploaded blob with uri " + blob.getUri() + " in " + getTime(endTime - startTime));
+    protected static String upload(TaskListener listener, CloudBlockBlob blob, FilePath src)
+			throws StorageException, IOException, InterruptedException, NoSuchAlgorithmException {
+		MessageDigest md = MessageDigest.getInstance("MD5");
```

Comment:
Can use `DigestUtils.getMd5Digest()` instead of `MessageDigest.getInstance("MD5")`